web/src/compose_actions.js

```diff
@@ -274,8 +274,7 @@ export function start(msg_type, opts) {
 
     if (opts.content !== undefined) {
         // If we were provided with message content, we might need to
-        // resize the compose box, or display that it's too long.
-        compose_ui.autosize_textarea($("textarea#compose-textarea"));
```

Comment:
I'm curious if we can merge the code in this `if (opts.content !== undefined) {` block with the one above, or if the stuff in the middle has to happen in between.

Reply:
> Are you removing this because it already happens in `insert_and_scroll_into_view`?

Yes, that's exactly why.

> I'm curious if we can merge the code

Thank you for bringing this to my attention! The 1st block could be merged into the 2nd, so I have done in a separate refactor commit.

Reply:
I think this change is wrong in the absence of the second commit, so it'd be better to do these changes in the opposite order.

`show_compose_box(msg_type, opts);` potentially triggers focus in the compose textarea, and if we haven't inserted the `content` yet, the cursor would end at the start, rather than the end.

Thinking about this caused me to realize that this change as a whole is also overriding the logic in `set_focus` for deciding which input to focus; if you for some reason don't have a recipient or topic yet but did have content (I'm not sure that's possible today, but can't prove it isn't), we'd be overriding that to focus the compose textarea.

It's possible we should be checking `get_focus_area` or otherwise reworking this; not sure.

Reply:
I rechecked the code, and after some testing I'm pretty sure that merging the 2nd block into the 1st (so that `show_compose_box` still comes after, and the `set_focus` logic is not overridden) works. 

If you draft a message with no topic, and then restore it, the cursor now spawns in the empty topic input as expected. For a draft with a complete recipient, the cursor appears at the end of the message content, scrolled into view if needed.